code/modules/mapping/mapping_helpers.dm

```diff
@@ -380,3 +381,33 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	log_mapping("[src] at [x],[y] could not find an airlock on current turf, cannot place paper note.")
 	qdel(src)
 
+//This helper applies traits to things on the map directly.
+/obj/effect/mapping_helpers/trait_injector
+	name = "Trait Injector"
+	icon_state = "trait"
+	late = TRUE
+	var/all = FALSE //Will inject into all fitting the criteria if true, otherwise first found
+	var/target_type //Will inject into atoms of this type
+	var/target_name //Will inject into atoms with this name
+	var/trait_name //Name of the trait, in NOT the define form
```

Comment:
```suggestion
	///Will inject into all fitting the criteria if true, otherwise first found.
	var/all = FALSE
	///Will inject into atoms of this type.
	var/target_type
	///Will inject into atoms with this name.
	var/target_name
	///Name of the trait, in the text (NOT the define) form.
	var/trait_name
```